### PR TITLE
Add paging to audit table entity

### PR DESCRIPTION
**Pull request by JayArora1997**
[US154265](https://rally1.rallydev.com/#/?detail=/userstory/705253427401&fdp=true): Course Merge - Audit Log UI- Paging

Related PR(s):
* https://github.com/Brightspace/ipsis-sis-course-merge/pull/152

### CODE REVIEW

**Review by kmcdaniel-d2l — APPROVED** [review]
LGTM, guess it would need a spot check on Course Merge page

File: src/ipsis/course-merge/CourseMergeOfferingBaseCollectionEntity.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class CourseMergeOfferingBaseCollectionEntity extends BaseCollectionEntity {
+
+	courseMergeOfferings() {
+		if (!this._entity) {
+			return;
+		}
+		return this._entity.entities;
+	}
+
+	loadMorePageSize() {
```
Comment (ckthiessen) [inline src/ipsis/course-merge/CourseMergeOfferingBaseCollectionEntity.js:12]: This method looks duplicated in `CourseMergeLogDetailCollectionEntity.js`. The only difference is the getter for the list length. Can you put this in `BaseCollectionEntity.js` and parameterize the getter?

File: src/ipsis/course-merge/BaseCollectionEntity.js [hunk 2]
```diff
@@ -20,17 +14,6 @@ export class BaseCollectionEntity extends Entity {
 		return this._pagingInfo()?.pageSize;
 	}
```
Comment (ckthiessen) [inline src/ipsis/course-merge/BaseCollectionEntity.js:16]:
```suggestion
loadMorePageSize(collectionGetter) {
		const pageSize = 20;
		const totalCount = this.totalCount() ?? 0;
		const courseMergeOfferingsLength = collectionGetter()?.length ?? 0;
		// if pageSize is larger than the number remaining items, return the number of remaining items to be loaded
		if (totalCount < courseMergeOfferingsLength + pageSize) {
			return totalCount - courseMergeOfferingsLength;
		}
		return pageSize;
	}
```

**Comment by JeremyWinkler** [comment]
Merge conflicts

**Review by zwilson2 — COMMENTED** [review]
Cole's comments and merge conflicts

File: src/ipsis/course-merge/CourseMergeOfferingBaseCollectionEntity.js (hunk 1 above)
Comment (JayArora1997) [reply src/ipsis/course-merge/CourseMergeOfferingBaseCollectionEntity.js:12]: Done!

File: src/ipsis/course-merge/BaseCollectionEntity.js (hunk 2 above)
Comment (JayArora1997) [reply src/ipsis/course-merge/BaseCollectionEntity.js:16]: Done!

File: src/ipsis/course-merge/BaseCollectionEntity.js [hunk 3]
```diff
@@ -62,5 +45,16 @@ export class BaseCollectionEntity extends Entity {
 	updateEntity(entity) {
 		this._entity = entity;
 	}
+
+	loadMorePageSize(collectionGetter) {
+		const pageSize = 20;
+		const totalCount = this.totalCount() ?? 0;
+		const collectionLength = collectionGetter?.length ?? 0;
```
Comment (zwilson2) [inline src/ipsis/course-merge/BaseCollectionEntity.js:52]: Your getters are functions
```suggestion
		const collectionLength = collectionGetter()?.length ?? 0;
```

**Review by zwilson2 — APPROVED** [review]

**Review by kmcdaniel-d2l — APPROVED** [review]

**Review by ckthiessen — APPROVED** [review]

**Review by kmcdaniel-d2l — APPROVED** [review]

**Pull request merged**